hwmv2: defconfig fixes

57300 opened the pull request:
Missing adaptions to HWMv2's defconfig inheritance scheme.

57300 committed:
boards: mps3: Fix non-secure variant

Both TRUSTED_EXECUTION_SECURE and TRUSTED_EXECUTION_NONSECURE were being
enabled, through a combination of multiple defconfigs.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
boards: xenvm: Unset HEAP_MEM_POOL_SIZE in gicv3 variant

In HWMv1, the `xenvm` and `xenvm_gicv3` boards used different heap sizes
- 16384 and 0 (default) respectively. Due to HWMv2 defconfig inheritance
the setting must be moved to `Kconfig.defconfig`.

As a result, `xenvm_xenvm_gicv3_defconfig` can be removed, because it is
now equal to `xenvm_defconfig` (base).

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
boards: qemu_cortex_a53: Fix Kconfig warnings in SMP variant

A combination of multiple defconfigs resulted in this:

   CONFIG_QEMU_ICOUNT=y
   CONFIG_QEMU_ICOUNT_SHIFT=4  # depends on QEMU_ICOUNT
   CONFIG_QEMU_ICOUNT_SLEEP=y  # depends on QEMU_ICOUNT
   CONFIG_QEMU_ICOUNT=n        # warning: unsatisfied dependencies

Fix this by setting the symbols in `Kconfig.defconfig` instead. Not only
does this allow for proper defconfig inheritance, it also allows samples
to control the value of QEMU_ICOUNT.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

nordicjm commented on boards/xenvm/xenvm_xenvm_gicv3_defconfig:
```diff
@@ -1,16 +0,0 @@
-# Enable UART driver
-CONFIG_SERIAL=y
```
Think this file is referenced in documentation file

57300 replied:
You're right, thanks. Pushed.

57300 committed:
boards: xenvm: doc: Remove reference to deleted file

`xenvm_xenvm_defconfig` no longer exists. Take this opportunity to refer
to board configurations by their actual identifiers.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

tejlmand commented on boards/xenvm/Kconfig.defconfig:
```diff
@@ -6,4 +6,7 @@ if BOARD_XENVM
 config BUILD_OUTPUT_BIN
 	default y
 
+config HEAP_MEM_POOL_SIZE
```
> Due to HWMv2 defconfig inheritance the setting must be moved to `Kconfig.defconfig`.

Technically this is a bit imprecise, as
- `xenvm_defconfig` could explicitly set  `CONFIG_HEAP_MEM_POOL_SIZE=16384`
- `xenvm_xenvm_gicv3_defconfig` could override by explicitly set `CONFIG_HEAP_MEM_POOL_SIZE=0`

That is possible because `KERNEL_MEM_POOL` is default `y`.

but as we now get rid of a `_defconfig` file, then this change is a bit nice.

57300 replied:
> * `xenvm_xenvm_gicv3_defconfig` could override by explicitly set `CONFIG_HEAP_MEM_POOL_SIZE=0`

True, I just didn't want to set a sticky value for it, because the original didn't.

tejlmand reviewed (approved):
Nice cleanup.

nordicjm commented:
@57300 needs rebase

nordicjm commented:
@57300 rebase needed
